IdentityCore/tests/MSIDMacKeychainTokenCacheTests.m

```diff
@@ -0,0 +1,354 @@
+                                                                                                   realm:accountB.realm
+                                                                                                    type:accountB.accountType];
+    keyA.username = accountA.username;
+    keyB.username = accountB.username;
+
+    BOOL result = [_cache saveAccount:accountA key:keyA serializer:_serializer context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    result = [_cache saveAccount:accountB key:keyB serializer:_serializer context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    MSIDAccountCacheItem* expectedAccount = accountB;
+    [expectedAccount setAdditionalAccountFields:@{@"key1": @"VALUE1", @"key2": @"value2", @"key3": @"VALUE3"}];
+
+    MSIDAccountCacheItem *actualAccount = [_cache accountWithKey:keyB
+                                                      serializer:_serializer
+                                                         context:nil
+                                                           error:&error];
+
+    XCTAssertNil(error);
+    XCTAssertNotNil(actualAccount);
+    XCTAssertTrue([expectedAccount isEqual:actualAccount]);
+
+    result = [_cache removeItemsWithAccountKey:keyB context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    // removing with keyB should delete the same keychain item referred to by keyA since they
+    // have the same primary key values
+    MSIDAccountCacheItem *deletedAccountA = [_cache accountWithKey:keyA
+                                                        serializer:_serializer
+                                                           context:nil
+                                                             error:&error];
+    XCTAssertNil(deletedAccountA);
+}
+
+- (void)testRemoveItemsWithAccountKey_whenKeyIsValid_shouldRemoveItem
+{
+    NSError *error;
+    BOOL result = [_cache saveAccount:_testAccount key:_testAccountKey serializer:_serializer context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    MSIDAccountCacheItem *account2 = [_cache accountWithKey:_testAccountKey
+                                                 serializer:_serializer
+                                                    context:nil
+                                                      error:&error];
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(_testAccount, account2);
+
+    result = [_cache removeItemsWithAccountKey:_testAccountKey context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    account2 = [_cache accountWithKey:_testAccountKey serializer:_serializer context:nil error:&error];
+    XCTAssertNil(error);
+    XCTAssertNil(account2);
+}
+
+- (void)testAccountsWithKey_whenAccountMissing_shouldReturnError
```

Comment:
nit: should it be "shouldNotReturnError"?

Reply:
fixed